stripe/src/main/java/com/stripe/android/StripeSuspend.kt

```diff
@@ -0,0 +1,682 @@
+ */
+@Throws(
+    AuthenticationException::class,
+    InvalidRequestException::class,
+    APIConnectionException::class,
+    APIException::class
+)
+suspend fun Stripe.retrieveSource(
+    @Size(min = 1) sourceId: String,
+    @Size(min = 1) clientSecret: String,
+    stripeAccountId: String? = this.stripeAccountId
+) = runOnIOAndThrowIfNull {
+    stripeRepository.retrieveSource(
+        sourceId,
+        clientSecret,
+        ApiRequest.Options(
+            apiKey = publishableKey,
+            stripeAccount = stripeAccountId
+        )
+    )
+}
+
+/**
+ * Suspend function to confirm a [SetupIntent] object.
+ *
+ * See [Confirm a SetupIntent](https://stripe.com/docs/api/setup_intents/confirm).
+ * `POST /v1/setup_intents/:id/confirm`
+ *
+ * @param confirmSetupIntentParams a set of params with which to confirm the Setup Intent
+ * @param idempotencyKey optional, see [Idempotent Requests](https://stripe.com/docs/api/idempotent_requests)
+ *
+ * @return a [SetupIntent] object
+ *
+ * @throws AuthenticationException failure to properly authenticate yourself (check your key)
+ * @throws InvalidRequestException your request has invalid parameters
+ * @throws APIConnectionException failure to connect to Stripe's API
+ * @throws APIException any other type of problem (for instance, a temporary issue with Stripe's servers)
+ */
+@Throws(
+    AuthenticationException::class,
+    InvalidRequestException::class,
+    APIConnectionException::class,
+    APIException::class
+)
+suspend fun Stripe.confirmSetupIntentSuspend(
+    confirmSetupIntentParams: ConfirmSetupIntentParams,
+    idempotencyKey: String? = null
+) = runOnIOAndThrowIfNull {
+    stripeRepository.confirmSetupIntent(
+        confirmSetupIntentParams,
+        ApiRequest.Options(
+            apiKey = publishableKey,
+            stripeAccount = stripeAccountId,
+            idempotencyKey = idempotencyKey
+        )
+    )
+}
+
+/**
+ * Suspend function to confirm a [PaymentIntent] object.
+ *
+ * See [Confirm a PaymentIntent](https://stripe.com/docs/api/payment_intents/confirm).
+ * `POST /v1/payment_intents/:id/confirm`
+ *
+ * @param confirmPaymentIntentParams a set of params with which to confirm the PaymentIntent
+ * @param idempotencyKey optional, see [Idempotent Requests](https://stripe.com/docs/api/idempotent_requests)
+ *
+ * @return a [PaymentIntent] object
+ *
+ * @throws AuthenticationException failure to properly authenticate yourself (check your key)
+ * @throws InvalidRequestException your request has invalid parameters
+ * @throws APIConnectionException failure to connect to Stripe's API
+ * @throws APIException any other type of problem (for instance, a temporary issue with Stripe's servers)
+ */
+@Throws(
+    AuthenticationException::class,
+    InvalidRequestException::class,
+    APIConnectionException::class,
+    APIException::class
+)
+suspend fun Stripe.confirmPaymentIntentSuspend(
```

Comment:
Same as above

Reply:
ditto

Reply:
Renamed to `Stripe#confirmPaymentIntent` after offline sync